aodndata/gsla/handler.py

```diff
@@ -0,0 +1,160 @@
+                if result_previous_version_creation_date > new_file_creation_date:
+                    push_new_file = False
+                else:
+                    remove_previous_version = True
+                    previous_file_path = self.get_previous_version_object(netcdf_file.name)
+
+            if push_new_file:
+                # index un-gzipped file, but push gzipped file to S3
+                netcdf_file.publish_type = PipelineFilePublishType.HARVEST_ONLY
+                # to set to all files in the collection
+                netcdf_file_gz_collection.set_publish_types(PipelineFilePublishType.UPLOAD_ONLY)
+
+            else:
+                netcdf_file.publish_type = PipelineFilePublishType.NO_ACTION
+                netcdf_file_gz_collection.publish_type = PipelineFilePublishType.NO_ACTION
+
+            # deletion of the previous file
+            if remove_previous_version:
+                previous_file_name = os.path.basename(previous_file_path)
+                file_to_delete = PipelineFile(previous_file_name,
+                                              is_deletion=True)
+                file_to_delete.dest_path = self.dest_path(previous_file_name)
+                file_to_delete.publish_type = PipelineFilePublishType.DELETE_ONLY
+                self.file_collection.add(file_to_delete)
+
+    def get_previous_version_object(self, filepath):
+        destination = self.dest_path(filepath)
+        destination_no_creation_date = re.sub('_C-[0-9]{8}T[0-9]{6}Z.*$', '', destination)
+
+        res = self.state_query.query_storage(destination_no_creation_date)
+        if len(res) > 1:
+            raise RuntimeError('More than 1 previous version of {filename} was found on storage.'.
+                               format(filename=os.path.basename(filepath)))
+        elif len(res) == 1 :
+            return res.keys()[0]
+        else:
+            return False
+
+    def get_previous_version_creation_date(self, filepath):
+        """
+        find matching old netcdf_file.gz object already on s3 by stripping the creation date of the new file object path
+        :return: old creation date if exists
+                 returns None if no previous file was found
+        """
+        prev = self.get_previous_version_object(filepath)
+        if prev:
+            return get_creation_date(prev)
+
+    @staticmethod
+    def dest_path(filepath):
+        file_basename = os.path.basename(filepath)
+        gsla_type = get_gsla_type(filepath)
+
+        if GSLA_REGEX_YEARLY.match(file_basename):
+            destination = os.path.join(GSLA_PREFIX_PATH, gsla_type, file_basename)
+        else:
+            fields = get_fields_from_filename(filepath, GSLA_REGEX)
+            gsla_year = datetime.strptime(fields['nc_time_cov_start'], '%Y%m%dT%H%M%SZ').year
+            destination = os.path.join(GSLA_PREFIX_PATH, gsla_type, str(gsla_year), file_basename)
+
+        # FORCE check we aren't deleted files that shouldn't be
```

Comment:
deleting

Reply:
This actually makes sure we don't delete what we shouldn't and  the following test tests this behaviour
```python
    @patch('aodndata.gsla.handler.GSLA_PREFIX_PATH', '')
    def test_setup_upload_location_push_newer_file_bad_prefix(self):
```
where this unittest rewrite the GSLA_PREFIX_PATH __IMOS/OceanCurrent__ to nothing

Reply:
I just meant to correct the typo in your comment from `deleted` to `deleting`